lib/cluster/cpg.c

```diff
@@ -848,13 +816,15 @@ pcmk__cpg_connect(pcmk_cluster_t *cluster)
     };
 
     cpg_evicted = false;
-    cluster->group.length = 0;
-    cluster->group.value[0] = 0;
+    cluster->priv->group.length = 0;
```

Comment:
SBD is broken now with this . Upon the existing call of crm_cluster_connect():

https://github.com/ClusterLabs/sbd/blob/42455eb4298fa4eae03330ca1a1bc373ece58c8f/src/sbd-cluster.c#L506

, sbd encounters segfault at here given that cluster->priv is not allocated.

Reply:
Thank you.

Sigh... That's because the connect functions assume an initialized cluster object. sbd doesn't call the constructor but rather uses a statically allocated `crm_cluster_t` object.

We could have `crm_cluster_connect()` allocate a `priv` member. I wouldn't want to do it in `pcmk_cluster_connect()`, since new callers should call it appropriately with an initialized object.

Of course there will be a trivial memory leak in sbd, since nothing will ever free the `priv` member. That's a small price to pay.

Reply:
Yes, sounds acceptable.

Reply:
https://github.com/ClusterLabs/pacemaker/pull/3548